aten/src/ATen/native/cuda/IndexKernel.cu

```diff
@@ -385,20 +382,9 @@ void masked_scatter_cuda_impl(Tensor& self, const Tensor& mask, const Tensor& so
   // Use a prefix sum to determine the output locations of the masked elements
   auto maskPrefixSum = at::empty_like(mask, mask.options().dtype(kLong));
 
-  auto allocator = THCThrustAllocator(globalContext().lazyInitCUDA());
-
-  thrust::device_ptr<mask_t> maskData(mask_cont.data_ptr<mask_t>());
-  thrust::device_ptr<int64_t> maskPrefixSumData(
-      maskPrefixSum.data_ptr<int64_t>());
-
-  // Reference for using static_cast on `init_value`:
-  // https://github.com/NVIDIA/thrust/issues/1379
-  thrust::exclusive_scan(
-      thrust::cuda::par(allocator).on(c10::cuda::getCurrentCUDAStream()),
-      maskData,
-      maskData + mask_cont.numel(),
-      maskPrefixSumData,
-      static_cast<int64_t>(0));
+  at::cuda::cub::exclusive_sum(
```

Comment:
mask_cont.numel() > int_max used to be supported w/o extravagant memory requirements, so probably we shouldn't remove this support? It's slightly annoying to add it (see cumsum where it's done for inclusive sum), but it's doable.